File: demos/jans-chip/ios/Jans-Chip/Models/DCResponse.swift
//
//  DCResponse.swift
//  Jans-Chip
//
//  Created by Nazar Yavornytskyi on 11.10.2023.
//

import Foundation

struct DCResponse: Codable {
    
    private enum CodingKeys: String, CodingKey {
        case clientId = "client_id"
        case clientSecret = "client_secret"
        case clientName = "client_name"
        case authorizationChallengeEndpoint = "authorization_challenge_endpoint"
        case endSessionEndpoint = "end_session_endpoint"
    }
    
    var clientId: String?
    var clientSecret: String?
    var clientName: String?
    var authorizationChallengeEndpoint: String?
    var endSessionEndpoint: String?
}

File: demos/jans-chip/android/app/src/main/java/io/jans/chip/model/DCResponse.kt
package io.jans.chip.model

import com.google.gson.annotations.SerializedName

data class DCResponse (
    @SerializedName("client_id")
    val clientId: String? = null,

    @SerializedName("client_secret")
    val clientSecret: String? = null,

    @SerializedName("client_name")
    val clientName: String? = null,

    @SerializedName("authorization_challenge_endpoint")
    val authorizationChallengeEndpoint: String? = null,

    @SerializedName("end_session_endpoint")
    val endSessionEndpoint: String? = null,

)